chains/orchestrator-relays/runtime/dancelight/src/bridge_to_ethereum_config.rs

```diff
@@ -345,6 +333,239 @@ where
+                "NativeContainerTokensProcessor::can_process_message: invalid gateway or channel"
+            );
+            return false;
+        }
+
+        // Try decoding the message and check if the destination owns the token being transferred
+        if let Some(token_data) =
+            NativeTokenTransferData::decode_native_token_message(&envelope.payload)
+        {
+            // Check if the asset is registered on EthereumSystem
+            if let Some(token_location) = Self::get_token_location(&token_data.token_id) {
+                Self::validate_destination_owns_token(&token_location, &token_data.destination)
+            } else {
+                log::error!(
+                    "NativeContainerTokensProcessor::can_process_message: token location not found for token_id: {:?}",
+                    token_data.token_id
+                );
+                false
+            }
+        } else {
+            false
+        }
+    }
+
+    fn process_message(_channel: Channel, envelope: Envelope) -> DispatchResult {
+        if let Some(token_data) =
+            NativeTokenTransferData::decode_native_token_message(&envelope.payload)
+        {
+            if let Some(token_location) = Self::get_token_location(&token_data.token_id) {
+                Self::process_native_token_transfer(token_data, token_location);
+                Ok(())
+            } else {
+                log::error!(
+                    "NativeContainerTokensProcessor::process_message: token location not found for token_id: {:?}",
+                    token_data.token_id
+                );
+                Ok(())
+            }
+        } else {
+            Ok(())
+        }
+    }
+}
+
+impl<T, EthereumLocation, EthereumNetwork, InboundQueuePalletInstance>
+    NativeContainerTokensProcessor<T, EthereumLocation, EthereumNetwork, InboundQueuePalletInstance>
+where
+    T: snowbridge_pallet_inbound_queue::Config
+        + pallet_ethereum_token_transfers::Config
+        + snowbridge_pallet_system::Config
+        + pallet_xcm::Config,
+    <T as frame_system::Config>::RuntimeEvent: From<RuntimeEvent>,
+    EthereumLocation: Get<Location>,
+    EthereumNetwork: Get<NetworkId>,
+    InboundQueuePalletInstance: Get<u8>,
+{
+    /// Gets the token location from a token_id using EthereumSystem pallet.
+    fn get_token_location(token_id: &H256) -> Option<Location> {
+        snowbridge_pallet_system::Pallet::<T>::convert(token_id)
+    }
+
+    /// Validates that the destination para_id owns the token being transferred.
+    fn validate_destination_owns_token(
+        token_location: &Location,
+        destination: &Destination,
+    ) -> bool {
+        let chain_part = token_location.interior().clone().split_global().ok();
+
+        let expected_para_id = match destination {
+            Destination::ForeignAccountId32 { para_id, .. } => *para_id,
+            Destination::ForeignAccountId20 { para_id, .. } => *para_id,
+            _ => 0u32,
```

Comment:
We might want to explicitly specify the default `para_id` to None or false, instead of defaulting it to a valid u32. And maybe process it accordingly.